packages/api-explorer/__tests__/JsonForm/index.test.jsx

```diff
@@ -46,60 +19,573 @@ describe('JSONForm ', () => {
+    const element = await mountJsonForm(props)
     const mockEvent = { preventDefault: jest.fn() }
     element.find('form').prop('onSubmit')(mockEvent)
     expect(props.onSubmit).toHaveBeenCalledTimes(1)
     expect(mockEvent.preventDefault).toHaveBeenCalledTimes(1)
   })
 
-  it('extend all json-editor editors', () => {
-    mountWithIntl(<JsonForm {...props} />)
-    expect(
-      extendMock.mock.calls.map(call => Object.keys(call[0]))
-    ).toEqual([
-      ['setContainer', 'build'], // foo
-      ['setContainer', 'build'], // multiple
-      ['setContainer', 'build'], // array - get-custom-editor
-      ['addControls', 'refreshValue'], // array-custom-editor  
-      ['setContainer', 'build'], // object - get-custom-editor
-      ['build'], // object-custom-editor
-      ['setContainer', 'build'], // not - get-custom-editor
-      ['preBuild'], // not-custom-editor
-      ['setContainer', 'build'], // anyOf - get-custom-editor
-      ['build', 'buildSwitcher', 'updateEditor', 'addErrorMessageHtmlNode',
-        'showErrorMessage', 'hideErrorMessage', 'insertNewEditor', 'hideEditor'] // anyOf-custom-editor
-    ])
+  it('on parsing error, shows error', async () => {
+    const element = await mountJsonForm({
+      ...props,
+      schema: {'$ref': '#/foo'}
+    })
+    expect(element.getDOMNode()).toMatchSnapshot()
+  })
+
+  it('converts schema correctly', async () => {
+    const schemaWithRef =  {
+      definitions: {
+        address: {
+          type: "object",
+          properties: {
+            street_address: { type: "string" },
+            city: { type: "string" },
+            state: { type: "string" }
+          },
+          required: ["street_address", "city", "state"]
+        }
+      },
+      $ref: "#/definitions/address"
+    };
+    const element = await mountJsonForm({...props, schema: schemaWithRef})
+    expect(element.find('JsonForm').state().jsonSchema).toEqual({
+      definitions: {
+        address: {
+          type: "object",
+          properties: {
+            street_address: { type: "string" },
+            city: { type: "string" },
+            state: { type: "string" }
+          },
+          required: ["street_address", "city", "state"]
+        }
+      },
+      type: "object",
+          properties: {
+            street_address: { type: "string" },
+            city: { type: "string" },
+            state: { type: "string" }
+          },
+          required: ["street_address", "city", "state"]
+    })
+  })
+
+  it('converts schema correctly with nested ref', async () => {
+    const element = await mountJsonForm({...props, schema: SCHEMA_BUG})
+    expect(element.find('JsonForm').state().jsonSchema).toEqual(EXPECTED_SCHEMA_BUG)
   })
 })
+
+function mountJsonForm (props) {
+  return new Promise((resolve) => {
+    const element = mountWithIntl(<JsonForm {...props} />)
+    setTimeout(() => {
+      element.update()
+      resolve(element)
+    }, 0)
+  })
+}
+
+const SCHEMA_BUG = {
```

Comment:
All these variables under test files could be testdata in some folder? I expect a folder/config123.json

This type of file could only increase